trunk/SUAVE/Components/Energy/Storages/Batteries/Constant_Mass/Lithium_Ion_LiFePO4_18650.py

```diff
@@ -0,0 +1,362 @@
+    
+            Assumptions:
+            None
+    
+            Source:
+            N/A
+    
+            Inputs:
+            state.unknowns.battery_voltage_under_load               [volts]
+    
+            Outputs: 
+            state.conditions.propulsion.battery_voltage_under_load  [volts]
+    
+            Properties Used:
+            N/A
+        """             
+        
+        segment.state.conditions.propulsion.battery_voltage_under_load  = segment.state.unknowns.battery_voltage_under_load
+        
+        return 
+    
+    def append_battery_residuals(self,segment,network): 
+        """ Packs the residuals specific to LFP cells to be sent to the mission solver.
+    
+            Assumptions:
+            None
+    
+            Source:
+            N/A
+    
+            Inputs:
+            state.conditions.propulsion:
+                motor_torque                          [N-m]
+                propeller_torque                      [N-m]
+                voltage_under_load                    [volts]
+            state.unknowns.battery_voltage_under_load [volts]
+            
+            Outputs:
+            None
+    
+            Properties Used:
+            network.voltage                           [volts]
+        """     
+        v_actual  = segment.state.conditions.propulsion.battery_voltage_under_load
+        v_predict = segment.state.unknowns.battery_voltage_under_load
+        v_max     = network.voltage
+        
+        # Return the residuals
+        segment.state.residuals.network.voltage = (v_predict[:,0] - v_actual[:,0])/v_max
+        
+        return 
+    
+    def append_battery_unknowns_and_residuals_to_segment(self,segment,initial_voltage, 
+                                              initial_battery_cell_temperature , initial_battery_state_of_charge,
+                                              initial_battery_cell_current,initial_battery_cell_thevenin_voltage): 
+        """ Sets up the information that the mission needs to run a mission segment using this network
+    
+            Assumptions:
+            None
+    
+            Source:
+            N/A
+    
+            Inputs:  
+            initial_voltage                       [volts]
+            initial_battery_cell_temperature      [Kelvin]
+            initial_battery_state_of_charge       [unitless]
+            initial_battery_cell_current          [Amperes]
+            initial_battery_cell_thevenin_voltage [Volts]
+            
+            Outputs
+            None
+            
+            Properties Used:
+            N/A
+        """        
+        
+        ones_row = segment.state.ones_row 
+        if initial_voltage==None:
+            initial_voltage = self.max_voltage 
+        segment.state.unknowns.battery_voltage_under_load  = initial_voltage * ones_row(1) 
+        
+        return  
+    
+    def compute_voltage(self,state):
```

Comment:
same comment; function already exists in base class

Reply:
Addressed